ex03.exs

```diff
@@ -59,8 +59,32 @@ defmodule Ex03 do
 
   """
 
+  defp split_check_size(n_splits, len) when n_splits > len do
+    raise ArgumentError, message: """
+      n_splits can't be greater than the length of the collection
+    """
+  end
+  defp split_check_size(_, _) do
+    :ok
+  end
+
```

Comment:
So... I gave some bonus points above. Now I'm going to take one away

## -1

This is kinda advanced stuff, and there's no real error here. It's more philosophical.

The comments were triggered by the fact that `split/2` below has lots of variables. I started wondering why, and realized it could be a lot simpler if there was no `check_split` function. 

So then I thought about `check_split`. It explicitly raises an exception if the process_count is greater than the collection size. But, really, what we _want_ to say is that `pmap` is not defined in those cases. 

If we assume the collection is a list, we could so this using

~~~ elixir
def pmap(collection, pc, func) when length(collection) >= pc do
~~~

`split` would then become:

``` elixir
def split(collection, n_splits) do
  chunk_size = collection |> Enum.count() |> div(n_splits)
  Enum.chunk_every(collection, chunk_size)
end
~~~

If, instead, `pmap` takes any collection, then it could be written as:

~~~
def pmap(collection, pc, func) do
  pmap_with_check(collection, Enum.count(collection), pc, func)
end

defp  pmap_with_check(collection, size, pc, func) when size >= pc do
...
~~~